test/TestProjects/MgmtSingleton/Generated/SingletonResource.cs

```diff
@@ -25,20 +33,30 @@ protected SingletonResource()
         /// <summary> Initializes a new instance of the <see cref = "SingletonResource"/> class. </summary>
         /// <param name="options"> The client parameters to use in these operations. </param>
         /// <param name="resource"> The resource that is the target of operations. </param>
-        internal SingletonResource(ArmResource options, SingletonResourceData resource) : base(options, ResourceIdentifier.RootResourceIdentifier)
+        internal SingletonResource(ArmResource options, SingletonResourceData resource) : base(options, resource.Id)
         {
             HasData = true;
             _data = resource;
+            _clientDiagnostics = new ClientDiagnostics(ClientOptions);
+            _restClient = new SingletonResourcesRestOperations(_clientDiagnostics, Pipeline, Id.SubscriptionId, BaseUri);
         }
 
         /// <summary> Initializes a new instance of the <see cref="SingletonResource"/> class. </summary>
         /// <param name="options"> The client parameters to use in these operations. </param>
-        internal SingletonResource(ArmResource options) : base(options, ResourceIdentifier.RootResourceIdentifier)
+        /// <param name="id"> The identifier of the resource that is the target of operations. </param>
+        internal SingletonResource(ArmResource options, ResourceIdentifier id) : base(options, id)
         {
+            _clientDiagnostics = new ClientDiagnostics(ClientOptions);
+            _restClient = new SingletonResourcesRestOperations(_clientDiagnostics, Pipeline, Id.SubscriptionId, BaseUri);
         }
 
-        /// <summary> Gets the parent resource of this resource. </summary>
-        public ArmResource Parent { get; }
```

Comment:
Since [SingletionOperations](https://github.com/Azure/azure-sdk-for-net/pull/23205/files#diff-fe1474cdf5b37bd020f25c35749b824f2f79d1019c148a96152581fd4e227b05) class is removed from core, we added [Parent](https://github.com/Azure/azure-sdk-for-net/blob/877d1e9dffc2b61c8a0f06a32e8b9ab8094fc260/sdk/resourcemanager/Azure.ResourceManager/src/SingletonOperations.cs#L31) property here.

CC: @m-nash

Reply:
We need to initialize this property in the ctors like https://github.com/Azure/azure-sdk-for-net/blob/877d1e9dffc2b61c8a0f06a32e8b9ab8094fc260/sdk/resourcemanager/Azure.ResourceManager/src/SingletonOperations.cs#L25

Reply:
Also, we need to initialize this property in the ctors like https://github.com/Azure/azure-sdk-for-net/blob/877d1e9dffc2b61c8a0f06a32e8b9ab8094fc260/sdk/resourcemanager/Azure.ResourceManager/src/SingletonOperations.cs#L25

Reply:
Yeah - I just realized that singleton resource is actually a combination of resource operations and resource container.
I will bring this back

Reply:
I have brought this back, please take a look again, thanks